libexec/bootstrap-scripts/main-dockerhub.sh

```diff
@@ -51,7 +51,7 @@ eval_abort "$SINGULARITY_libexecdir/singularity/python/import.py"
 for i in `cat "$SINGULARITY_CONTENTS"`; do
     name=`basename "$i"`
     message 1 "Exploding layer: $name\n"
-    zcat "$i" | (cd "$SINGULARITY_ROOTFS"; tar --exclude=dev/* -xf -) || exit $?
+    zcat "$i" | (cd "$SINGULARITY_ROOTFS"; tar --exclude=dev/* --exclude=*/.wh.* -xf -) || exit $?
```

Comment:
I have a small worry that there may be other file(s) that match the */.wh.* pattern that are not whiteout files.  Is there any other way to identify them?  How does aufs identify whiteout files?  Pattern matching alone?

Reply:
https://github.com/torvalds/linux/blob/master/Documentation/filesystems/overlayfs.txt#L84-L99